# EDA en GRD, parte 3

Continuación del analizan los datos del Conjunto Mínimo Básico de Datos (CMBD) 2023, obtenidos desde FONASA.

Fuente: https://www.fonasa.cl/sites/fonasa/datos-abiertos/bases-grd.

[Link al archivo GRD_PUBLICO_2023.csv](https://drive.google.com/file/d/1jzk5VbuMZbnBNZ8AmIFeBVPrg0TFajgM/view?usp=drive_link//)

El Archivo Tablas maestras base GRD, contiene los codigos y nombres de los Hospitales, permitirá cruzar la infromación para obtener el nombre del Hospital, ya que el archivo no lo contiene.

[Link al archivo Tablas maestras base GRD](https://docs.google.com/spreadsheets/d/1hg-zSNNGpCHGCyWyn7dQEojlNH9xd5zp/edit?usp=sharing&ouid=104355169187812183586&rtpof=true&sd=true)

El archivo GRE DESC en formato excel contiene la descripcion del GRD con el código numérico correspondiente, permitiendo el cruce de información para agregar al dataset.

[Link al archivo GRD DESC](https://docs.google.com/spreadsheets/d/1LwMJCcihsfTwyWfC6ngop2bQS5gk1dNG/edit?usp=sharing&ouid=105597216880616359423&rtpof=true&sd=true)

##Carga, limpieza y descripción de los datos

Lo primero es importar las librerias que se utilizaran en el procesamiento del set de datos. Estas librerias pueden importarse en cualquier momento o lugar del código.

En este caso colab utiliza módulos, que son los rectángulos donde se programa, entonces al correr el código por modulo se pueden apreciar de mejor forma los errores que se pueden ir cometiendo.

In [1]:
# Importar librerias
# Anteponer el signo gato '#' para escribir sin que afecte el código.

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import math

In [2]:
# Cargar un archivo csv
df = pd.read_csv('GRD_PUBLICO_2023.csv', encoding='latin-1', delimiter=';')

<ipython-input-2-986744a3f4aa>:2: DtypeWarning: Columns (24,25,26,27,28,29,30,31,32,33,43,47,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,105,128) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('GRD_PUBLICO_2023.csv', encoding='latin-1', delimiter=';')


In [3]:
# Filas y Columnas
# df.shape

In [4]:
# Código para ver todas las columnas
pd.set_option('display.max_columns', None)

In [5]:
# Conocer el tipo de dato que hay en las primeras 10 columnas
# df.dtypes.head(10)

In [6]:
# ver el nombre de todas las columnas
print(df.columns.tolist())

['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA', 'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD', 'TIPO_PROCEDENCIA', 'TIPO_INGRESO', 'ESPECIALIDAD_MEDICA', 'TIPO_ACTIVIDAD', 'FECHA_INGRESO', 'SERVICIOINGRESO', 'FECHATRASLADO1', 'SERVICIOTRASLADO1', 'FECHATRASLADO2', 'SERVICIOTRASLADO2', 'FECHATRASLADO3', 'SERVICIOTRASLADO3', 'FECHATRASLADO4', 'SERVICIOTRASLADO4', 'FECHATRASLADO5', 'SERVICIOTRASLADO5', 'FECHATRASLADO6', 'SERVICIOTRASLADO6', 'FECHATRASLADO7', 'SERVICIOTRASLADO7', 'FECHATRASLADO8', 'SERVICIOTRASLADO8', 'FECHATRASLADO9', 'SERVICIOTRASLADO9', 'FECHAALTA', 'SERVICIOALTA', 'TIPOALTA', 'CONDICIONDEALTANEONATO1', 'PESORN1', 'SEXORN1', 'RN1ESTADO', 'CONDICIONDEALTANEONATO2', 'PESORN2', 'SEXORN2', 'RN2ESTADO', 'CONDICIONDEALTANEONATO3', 'PESORN3', 'SEXORN3', 'RN3ESTADO', 'CONDICIONDEALTANEONATO4', 'PESORN4', 'SEXORN4', 'RN4ESTADO', 'DIAGNOSTICO1', 'DIAGNOSTICO2', 'DIAGNOSTICO3', 'DIAGNOSTICO4', 'DIAGNOSTICO5', 'DIAGNOSTICO6', 'DIAGNOST

Exiten 129 columnas en el archivo

## Cruce entre archivos

Los nombres de los Hospitales estan dentro de los archivo de la pagina de FONASA de Datos Abiertos

El archivo excel: 'Tablas maestras bases GRD.xlsx' cotiene los codigos y los nombres de los hospitales.

Los siguientes codigos cruzan la información y agregan una columna al lado del codigo del hospital con su respectivo nombre.

Luego se utiliza nunique() que entrega la cantidad de valores unicos encontrados, en este caso, en la columna 'COD_HOSPITAL'.

Para terminar esta parte se utiliza value_counts() en la columna 'NOMBRE_HOSPITAL', que devuelve la cantidad de valores contenidos por cada hospital. Este valor no es preciso, ya que aún no se han revisado los errores que puede contener este set de datos.

In [7]:
# cargar el archivo excel que cruzaremos con el data set

hospitales_df = pd.read_excel('Tablas maestras bases GRD.xlsx', sheet_name='Hospitales')

In [8]:
# Ver el df que creamos desde el archivo excel.
hospitales_df.head()

HOSPITALES                              Unnamed: 1
0      101100  Hospital Dr. Juan Noé Crevanni (Arica)
1      101203                            Clínica Hebe
2      101212                     Clínica San Agustín
3      101213                        Clínica San José
4      101215                  Complejo Penitenciario

In [9]:
# A la columna Unnamed 1 de la lista de hospitales poner el nombre: NOMBRE_HOSPITAL
hospitales_df = hospitales_df.rename(columns={'Unnamed: 1': 'NOMBRE_HOSPITAL'})

In [10]:
#
hospitales_df = hospitales_df.rename(columns={'HOSPITALES': 'COD_HOSPITAL'})
df = pd.merge(df, hospitales_df[['COD_HOSPITAL', 'NOMBRE_HOSPITAL']], on='COD_HOSPITAL', how='inner') #Usamos on porque las columnas se llaman igual

In [11]:
# Mover la columna 'NOMBRE_HOSPITAL' a la derecha de la columna 'COD_HOSPITAL'
cols = list(df.columns)
cols.insert(cols.index('COD_HOSPITAL') + 1, cols.pop(cols.index('NOMBRE_HOSPITAL')))
df = df[cols]

Un detalle imporante es el número a la izquierda de la primera columna llamado index o índice

In [13]:
df.index = df.index + 1

In [15]:
# cantidad de valores en COD_HOSPITAL
df['COD_HOSPITAL'].nunique()

68

Hay 68 valores distintos en COD_HOSPITAL lo que refleja que existen datos de 68 Hospitales en este data set.

In [16]:
# ver todas las filas
pd.set_option('display.max_rows', None)

In [17]:
# count value para COD_HOSPITAL
df['NOMBRE_HOSPITAL'].value_counts()

NOMBRE_HOSPITAL
Complejo Hospitalario Dr. Sótero del Río (Santiago, Puente Alto)            49653
Hospital Clínico Regional Dr. Guillermo Grant Benavente (Concepción)        34144
Hospital Dr. César Garavagno Burotto (Talca)                                32148
Complejo Asistencial Dr. Víctor Ríos Ruiz (Los Ángeles)                     31251
Hospital Dr. Hernán Henríquez Aravena (Temuco)                              29786
Hospital de Puerto Montt                                                    29283
Complejo Hospitalario San José (Santiago, Independencia)                    28496
Hospital San Juan de Dios (Santiago, Santiago)                              26745
Hospital Barros Luco Trudeau (Santiago, San Miguel)                         26701
Hospital Dr. Gustavo Fricke (Viña del Mar)                                  26624
Hospital Regional de Rancagua                                               26257
Hospital Dr. Félix Bulnes Cerda (Santiago, Quinta Normal)                   25063
Hospital Clínico Herminda Martín (Chillán)                                  25052
Hospital Clínico Regional (Valdivia)                                        23837
Hospital Las Higueras (Talcahuano)                                          23691
Hospital Clínico Metropolitano La Florida Dra. Eloisa Díaz Inzunza          21176
Hospital Dr. Leonardo Guzmán (Antofagasta)                                  21076
Hospital Carlos Van Buren (Valparaíso)                                      20843
Hospital Dr. Ernesto Torres Galdames (Iquique)                              20603
Hospital Clínico Metropolitano El Carmen Doctor Luis Valentín Ferrada       20415
Hospital San Juan de Dios (La Serena)                                       19951
Hospital Clínico San Borja-Arriarán (Santiago, Santiago)                    19812
Hospital Dr. Juan Noé Crevanni (Arica)                                      19420
Hospital Base San José de Osorno                                            19407
Hospital Padre Alberto Hurtado (San Ramón)                                  19194
Hospital San Pablo (Coquimbo)                                               18575
Hospital Del Salvador (Santiago, Providencia)                               18308
Hospital San Juan de Dios (Curicó)                                          18087
Hospital Dr. Luis Tisné B. (Santiago, Peñalolén)                            17553
Hospital El Pino (Santiago, San Bernardo)                                   15818
Hospital Presidente Carlos Ibáñez del Campo (Linares)                       14567
Hospital San Martín (Quillota)                                              13728
Hospital San José del Carmen (Copiapó)                                      13600
Hospital Clínico de Magallanes Dr. Lautaro Navarro Avaria                   13195
Hospital Dr. Antonio Tirado Lanas (Ovalle)                                  12687
Hospital de San Camilo (San Felipe)                                         12633
Hospital Dr. Carlos Cisternas (Calama)                                      11384
Hospital de Urgencia Asistencia Pública Dr. Alejandro del Río               11238
Hospital Dr. Mauricio Heyermann (Angol)                                     10343
Hospital Regional (Coihaique)                                                9777
Hospital San Juan de Dios (San Fernando)                                     9483
Hospital Claudio Vicuña (San Antonio)                                        9338
Hospital Dr. Exequiel González Cortés (Santiago, San Miguel)                 9079
Hospital de Quilpué                                                          8998
Hospital San José (Melipilla)                                                8895
Hospital de Niños Dr. Luis Calvo Mackenna (Santiago, Providencia)            8816
Hospital Clínico de Niños Dr. Roberto del Río (Santiago, Independencia)      8597
Hospital de San Carlos                                                       8500
Hospital de Castro                              

## Post 2: Valores Nulos y Ceros

Una vez detectados los valores nulos, faltantes, erroneos o que contengan 0 donde no corresponda, viene la decision de que hacer con estas filas dependiendo de con que datos estemos trabajando.

Estos datos faltantes o ceros los podemos reemplazar por el valor promedio, eliminar, rellenar con 0, entre otras opciones.

### isin([0])  y isna:

Son formulas que permiter conocer los valores nulos y ceros que contengan las columnas del data frame.

- .isna() detecta valores faltantes (NaN, NA, etc.) en un DataFrame o Serie de pandas.
- .isin([0]) busca valores específicamente iguales a 0.

In [18]:
# valores faltantes en df
# El código nos entrega la suma de los valores nulos o faltantes para cada columna
df.isna().sum()

COD_HOSPITAL                      0
NOMBRE_HOSPITAL                   0
CIP_ENCRIPTADO                    0
SEXO                              0
FECHA_NACIMIENTO                 10
ETNIA                             0
PROVINCIA                         0
COMUNA                            0
NACIONALIDAD                      0
PREVISION                         0
SERVICIO_SALUD                    0
TIPO_PROCEDENCIA                  0
TIPO_INGRESO                      0
ESPECIALIDAD_MEDICA               0
TIPO_ACTIVIDAD                    0
FECHA_INGRESO                     0
SERVICIOINGRESO                   0
FECHATRASLADO1               872748
SERVICIOTRASLADO1            872748
FECHATRASLADO2               984495
SERVICIOTRASLADO2            984495
FECHATRASLADO3              1023334
SERVICIOTRASLADO3           1023334
FECHATRASLADO4              1033767
SERVICIOTRASLADO4           1033767
FECHATRASLADO5              1037308
SERVICIOTRASLADO5           1037308
FECHATRASLADO6              1038566
SERVICIOTRASLADO6           1038566
FECHATRASLADO7              1039074
SERVICIOTRASLADO7           1039074
FECHATRASLADO8              1039285
SERVICIOTRASLADO8           1039285
FECHATRASLADO9              1039408
SERVICIOTRASLADO9           1039408
FECHAALTA                         0
SERVICIOALTA                      0
TIPOALTA                          0
CONDICIONDEALTANEONATO1     1039587
PESORN1                      908370
SEXORN1                      932854
RN1ESTADO                    931844
CONDICIONDEALTANEONATO2     1039587
PESORN2                     1038192
SEXORN2                     1038192
RN2ESTADO                   1038192
CONDICIONDEALTANEONATO3     1039587
PESORN3                     1039574
SEXORN3                     1039574
RN3ESTADO                   1039574
CONDICIONDEALTANEONATO4     1039587
PESORN4                     1039587
SEXORN4                     1039587
RN4ESTADO                   1039587
DIAGNOSTICO1                      0
DIAGNOSTICO2                 133709
DIAGNOSTICO3                 268293
DIAGNOSTICO4                 398948
DIAGNOSTICO5                 525141
DIAGNOSTICO6                 633878
DIAGNOSTICO7                 722949
DIAGNOSTICO8                 794214
DIAGNOSTICO9                 849815
DIAGNOSTICO10                892311
DIAGNOSTICO11                924776
DIAGNOSTICO12                949542
DIAGNOSTICO13                968718
DIAGNOSTICO14                983619
DIAGNOSTICO15                995137
DIAGNOSTICO16               1004218
DIAGNOSTICO17               1011310
DIAGNOSTICO18               1016928
DIAGNOSTICO19               1021377
DIAGNOSTICO20               1025016
DIAGNOSTICO21               1027879
DIAGNOSTICO22               1030060
DIAGNOSTICO23               1031856
DIAGNOSTICO24               1033338
DIAGNOSTICO25               1034504
DIAGNOSTICO26               1035480
DIAGNOSTICO27               1036236
DIAGNOSTICO28               1036837
DIAGNOSTICO29               1037374
DIAGNOSTICO30               1037793
DIAGNOSTICO31               1038142
DIAGNOSTICO32               1038424
DIAGNOSTICO33               1038643
DIAGNOSTICO34               1038838
DIAGNOSTICO35               1039065
PROCEDIMIENTO1                  172
PROCEDIMIENTO2                83825
PROCEDIMIENTO3               174514
PROCEDIMIENTO4               257955
PROCEDIMIENTO5               336209
PROCEDIMIENTO6               413438
PROCEDIMIENTO7               490979
PROCEDIMIENTO8               567313
PROCEDIMIENTO9               640063
PROCEDIMIENTO10              705976
PROCEDIMIENTO11              763341
PROCEDIMIENTO12              810767
PROCEDIMIENTO13              849814
PROCEDIMIENTO14              881287
PROCEDIMIENTO15              906292
PROCEDIMIENTO16              926712
PROCEDIMIENTO17              943480
PROCEDIMIENTO18              957010
PROCEDIMIENTO19              968399
PROCEDIMIENTO20              977905
PROCEDIMIENTO21              985904
PROCEDIMIENTO22              992761
PROC

Llaman la atención los valores nulos de las siguientes columnas, con la cantidad de filas correspondiente.

- FECHA_NACIMIENTO:    10
- PROCEDIMIENTO1:      172
- IR_29301_COD_GRD:    30
- IR_29301_PESO:       30
- IR_29301_SEVERIDAD:  30
- IR_29301_MORTALIDAD: 30

### Columna FECHA_NACIMIENTO

In [19]:
# conocer la cantidad de filas o casos que contienen valores nulos en la columna 'FECHA_NACIMIENTO'
df['FECHA_NACIMIENTO'].isna().sum()

10

Hay 10 casos fechas de nacimiento nulas o faltantes

In [20]:
# Mostrar las filas con valores nulos para la columna FECHA_NACIMIENTO
df[df['FECHA_NACIMIENTO'].isnull()]

COD_HOSPITAL                                    NOMBRE_HOSPITAL  \
109350        113180          Hospital El Pino (Santiago, San Bernardo)   
139601        107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
276986        111101  Hospital Clínico Metropolitano El Carmen Docto...   
466554        115107           Hospital San Juan de Dios (San Fernando)   
501641        129106                       Hospital San José (Victoria)   
548620        115100                      Hospital Regional de Rancagua   
555661        111101  Hospital Clínico Metropolitano El Carmen Docto...   
793625        111101  Hospital Clínico Metropolitano El Carmen Docto...   
913560        118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
968355        115100                      Hospital Regional de Rancagua   

       CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO ETNIA   PROVINCIA     COMUNA  \
109350       96389959   MUJER              NaN  OTRO    SANTIAGO   SANTIAGO   
139601       73185594   MUJER              NaN  OTRO    QUILLOTA   QUILLOTA   
276986       99656404   MUJER              NaN  OTRO    SANTIAGO      MAIPU   
466554       85212242   MUJER              NaN  OTRO   COLCHAGUA   PLACILLA   
501641       93147538   MUJER              NaN  OTRO     MALLECO   VICTORIA   
548620       73630266  HOMBRE              NaN  OTRO   CACHAPOAL   RANCAGUA   
555661       99646346   MUJER              NaN  OTRO    SANTIAGO      MAIPU   
793625       99707489   MUJER              NaN  OTRO    SANTIAGO  CERRILLOS   
913560       67821989   MUJER              NaN  OTRO  CONCEPCION    CORONEL   
968355       75146701   MUJER              NaN  OTRO   COLCHAGUA   NANCAGUA   

                                NACIONALIDAD                       PREVISION  \
109350                                  CUBA  FONASA INSTITUCIONAL - (MAI) C   
139601                                 CHILE  FONASA INSTITUCIONAL - (MAI) B   
276986  VENEZUELA (REPÚBLICA BOLIVARIANA DE)  FONASA INSTITUCIONAL - (MAI) A   
466554                                 CHILE  FONASA INSTITUCIONAL - (MAI) B   
501641                                 CHILE  FONASA INSTITUCIONAL - (MAI) C   
548620                                 CHILE  FONASA INSTITUCIONAL - (MAI) B   
555661                                  PERÚ  FONASA INSTITUCIONAL - (MAI) A   
793625     BOLIVIA (ESTADO PLURINACIONAL DE)  FONASA INSTITUCIONAL - (MAI) A   
913560                                 CHILE  FONASA INSTITUCIONAL - (MAI) A   
968355                                 CHILE  FONASA INSTITUCIONAL - (MAI) B   

                 SERVICIO_SALUD  \
109350        METROPOLITANO SUR   
139601    VIÑA DEL MAR QUILLOTA   
276986    METROPOLITANO CENTRAL   
466554  LIBERTADOR B. O HIGGINS   
501641          ARAUCANÍA NORTE   
548620  LIBERTADOR B. O HIGGINS   
555661    METROPOLITANO CENTRAL   
793625    METROPOLITANO CENTRAL   
913560               CONCEPCIÓN   
968355  LIBERTADOR B. O HIGGINS   

                                         TIPO_PROCEDENCIA TIPO_INGRESO  \
109350                    SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA   
139601                         OTROS HOSPITALES DE LA RED   PROGRAMADA   
276986                    SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA   
466554                    SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA   
501641  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
548620  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
555661                    SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA   
793625                    SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA   
913560  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
968355  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   

              ESPECIALIDAD_MEDICA                   TIPO_ACTIVIDAD  \
109350  OBSTETRICIA Y GINECOLOGÍA                  HOSPITALIZACIÓN   
139601     CIRUGÍA CARDIOVASCULAR                  HOSPITALIZACIÓN   
276986           MED

Podemos visualizar en la columna o variable FECHA_NACIMIENTO que los valores faltantes se representan con NaN

### Columna PROCEDIMIENTO1

In [21]:
# ver los datos donde PROCEDIMIENTO1 contiene valores nulos
df[df['PROCEDIMIENTO1'].isna()]

COD_HOSPITAL                                    NOMBRE_HOSPITAL  \
3455           118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
4807           111101  Hospital Clínico Metropolitano El Carmen Docto...   
6396           118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
15355          107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
29300          118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
36100          107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
36811          107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
37534          118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
46276          111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
55224          118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
59333          107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
60954          111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
69714          118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
72857          111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
83801          107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
92029          111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
93384          118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
99799          118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
103545         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
103820         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
111410         118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
116498         110150                      Hospital San José (Melipilla)   
119363         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
120744         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
121338         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
121407         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
127541         114103         Hospital Padre Alberto Hurtado (San Ramón)   
146352         124105                           Hospital de Puerto Montt   
149716         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
167228         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
185038         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
189535         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
190018         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
200133         129106                       Hospital San José (Victoria)   
204326         118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
206564         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
212400         109101  Hospital Clínico de Niños Dr. Roberto del Río ...   
220933         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
230282         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
252278         118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
253462         105100              Hospital San Juan de Dios (La Serena)   
271137         103100         Hospital Dr. Leonardo Guzmán (Antofagasta)   
277288         118100  Hospital Clínico Regional Dr. Guillermo Grant ...   
280033         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
289490         107100         Hospital Dr. Gustavo Fricke (Viña del Mar)   
294798         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
299475         124105                           Hospital de Puerto Montt   
303038         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
311325         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
311852         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
314352         111100  Hospital Clínico San Borja-Arriarán (Santiago,...   
315349         118100  Hospital Clínico Regional Dr. Guil

###Analisis de PROCEDIMIENTO1 y PROCEDIMIENTO2

Al revisar la columna PROCEDIMIENTO1 con valores nulos se aprecia en la columna PROCEDIMIENTO2 que hay algunos valores en esta última.

El siguiente código entrega los valores con tenidos en PROCEDIMIENTO2 pero en las filas que contienen nulos en PROCEDIMIENTO1

In [22]:
# df[df['PROCEDIMIENTO1'].isna()] con valores nulos para PROCEDIMIENTO2
df[df['PROCEDIMIENTO1'].isna()]['PROCEDIMIENTO2'].value_counts()

PROCEDIMIENTO2
8703.0    1
8622.0    1
8867.0    1
8879.0    1
4467.0    1
5587.0    1
8847.0    1
9059.0    1
8856.0    1
Name: count, dtype: int64

El código nos permite ver los dígitos CIE9 utilizados en PROCEDIMIENTO2, arrojando 9 filas o casos que estan dentro de las 172 columnas con valores nulos de PROCEDIMIENTO1.

Esto quiere decir que hay 9 casos con un error que llama la atención, ya que se registraron 9 codigo en PROC2 pero no en PROC1.

In [23]:
# En df[df['PROCEDIMIENTO1'].isna()] ver los valores nulos para PROCEDIMIENTO2
df[df['PROCEDIMIENTO1'].isna()]['PROCEDIMIENTO2'].isna().sum()

163

el código .Isna siempre nos devuelve los valores nulos.

Al combinarlo con sum, son entrega la cantidad de valores nulos. Son 163 de 172 valores nulos, por lo que hay 9 filas que tienen digitos CIE9. Esto corrobora la información de más arriba.



### Columna IR_29301_COD_GRD

In [24]:
# visualizar donde 'IR_29301_COD_GRD' contiene valores nulos
df[df['IR_29301_COD_GRD'].isna()]

COD_HOSPITAL                                    NOMBRE_HOSPITAL  \
32678          114105  Hospital Clínico Metropolitano La Florida Dra....   
211909         114105  Hospital Clínico Metropolitano La Florida Dra....   
245163         114105  Hospital Clínico Metropolitano La Florida Dra....   
257735         114105  Hospital Clínico Metropolitano La Florida Dra....   
271714         114105  Hospital Clínico Metropolitano La Florida Dra....   
273577         114105  Hospital Clínico Metropolitano La Florida Dra....   
290395         114105  Hospital Clínico Metropolitano La Florida Dra....   
372316         114105  Hospital Clínico Metropolitano La Florida Dra....   
372513         114105  Hospital Clínico Metropolitano La Florida Dra....   
391376         114105  Hospital Clínico Metropolitano La Florida Dra....   
406496         114105  Hospital Clínico Metropolitano La Florida Dra....   
422609         114105  Hospital Clínico Metropolitano La Florida Dra....   
498700         114105  Hospital Clínico Metropolitano La Florida Dra....   
501552         114105  Hospital Clínico Metropolitano La Florida Dra....   
515480         114105  Hospital Clínico Metropolitano La Florida Dra....   
576036         114105  Hospital Clínico Metropolitano La Florida Dra....   
628769         114105  Hospital Clínico Metropolitano La Florida Dra....   
789947         114105  Hospital Clínico Metropolitano La Florida Dra....   
793132         114105  Hospital Clínico Metropolitano La Florida Dra....   
810590         114105  Hospital Clínico Metropolitano La Florida Dra....   
832939         114105  Hospital Clínico Metropolitano La Florida Dra....   
866890         114105  Hospital Clínico Metropolitano La Florida Dra....   
881962         114105  Hospital Clínico Metropolitano La Florida Dra....   
888313         114105  Hospital Clínico Metropolitano La Florida Dra....   
931290         114105  Hospital Clínico Metropolitano La Florida Dra....   
941602         114105  Hospital Clínico Metropolitano La Florida Dra....   
943834         114105  Hospital Clínico Metropolitano La Florida Dra....   
948985         114105  Hospital Clínico Metropolitano La Florida Dra....   
973956         114105  Hospital Clínico Metropolitano La Florida Dra....   
1032798        114105  Hospital Clínico Metropolitano La Florida Dra....   

        CIP_ENCRIPTADO    SEXO FECHA_NACIMIENTO ETNIA PROVINCIA      COMUNA  \
32678         74189482   MUJER       09-03-1995  OTRO  SANTIAGO  LA FLORIDA   
211909        72640269  HOMBRE       07-05-1956  OTRO  SANTIAGO  LA FLORIDA   
245163        81462281  HOMBRE       12-04-1978  OTRO  SANTIAGO  LA FLORIDA   
257735        78149802  HOMBRE       22-01-1959  OTRO  SANTIAGO  LA FLORIDA   
271714        77084522  HOMBRE       04-09-1984  OTRO  SANTIAGO  LA FLORIDA   
273577        74334525   MUJER       21-09-1990  OTRO  SANTIAGO  LA FLORIDA   
290395        74992858   MUJER       05-01-1956  OTRO  SANTIAGO  LA PINTANA   
372316        69770494   MUJER       14-11-1969  OTRO  SANTIAGO  LA FLORIDA   
372513        95923094   MUJER       09-11-1995  OTRO  SANTIAGO  LA FLORIDA   
391376        69726942  HOMBRE       11-06-1957  OTRO  SANTIAGO  LA FLORIDA   
406496        76672175  HOMBRE       11-05-1961  OTRO  SANTIAGO  LA FLORIDA   
422609        72107277   MUJER       03-11-1983  OTRO  SANTIAGO  LA FLORIDA   
498700        82302566  HOMBRE       11-02-1990  OTRO  SANTIAGO  LA FLORIDA   
501552        71041177   MUJER       26-03-1964  OTRO  SANTIAGO   LA GRANJA   
515480        74848098  HOMBRE       12-10-1948  OTRO  SANTIAGO  LA FLORIDA   
576036        78687273   MUJER       31-03-1990  OTRO  SANTIAGO  LA FLORIDA   
628769        77959272   MUJER       12-10-1945  OTRO  SANTIAGO  LA FLORIDA   
789947        72373696  HOMBRE       18-03-1954  OTRO  SANTIAGO  LA FLORIDA   
793132        72164454  HOMBRE       14-04-1937  OTRO  SANTIAGO  LA FLORIDA   
810590        73191779   MUJER       23-06-1988  OTRO  SANTIAGO  LA FLOR

Los 30 valores nulos en la columna GRD pertenecen al mismo hospital y contienen el valor DESCONOCIDO en la columna SERVICIOALTA.

In [25]:
# df['SERVICIOALTA'] cantidad de valores distintos o unicos
df['SERVICIOALTA'].nunique()

87

In [26]:
# count.value para la columna SERVICIOALTA
df['SERVICIOALTA'].value_counts()

SERVICIOALTA
UNIDAD DE RECUPERACIÓN DE PABELLONES (CENTRAL Y CMA)              129020
CIRUGÍA                                                           107584
MEDICINA                                                          103103
OBSTETRICIA                                                        79049
PEDIATRÍA                                                          52653
GINECOLOGÍA                                                        42667
HOSPITAL DE DÍA QUIRÚRGICO                                         37990
AREA MEDICO-QUIRURGICO CUIDADOS MEDIOS                             36941
TRAUMATOLOGÍA Y ORTOPEDIA                                          34514
OBSTETRICIA Y GINECOLOGÍA                                          33542
AREA MEDICO-QUIRURGICO CUIDADOS BASICOS                            22427
CIRUGÍA INFANTIL                                                   21398
AREA QUIRÚRGICA                                                    19175
PUERPERIO                                                          19055
AGUDOS MEDICINA                                                    16150
UNIDAD DE TRATAMIENTO INTERMEDIO (UTI) (INDIFERENCIADO) ADULTO     15264
AREA MEDICA ADULTO CUIDADOS MEDIOS                                 14988
DERIVACIÓN MÉDICO QUIRÚRGICO                                       13275
PENSIONADO ADULTO                                                  13035
UROLOGÍA                                                           13028
OFTALMOLOGÍA                                                       12400
NEONATOLOGÍA CUNAS                                                 11467
AREAOBSTETRICA                                                     11256
AREA MÉDICA                                                        10101
UNIDAD DE TRATAMIENTO INTERMEDIO NEONATOLOGÍA                       9035
NEUROCIRUGÍA ADULTO                                                 8946
LACTANTES                                                           8004
AREA MEDICA ADULTO CUIDADOS BASICOS                                 7709
ONCOLOGÍA                                                           7653
UNIDAD DE TRATAMIENTO INTERMEDIO CORONARIO                          7214
EMBARAZO ALTO RIESGO OBSTÉTRICO (PATOLÓGICO)                        6776
DESCONOCIDO                                                         6534
UNIDAD DE CUIDADOS INTENSIVOS ADULTO                                6449
SEGUNDA INFANCIA                                                    6019
UNIDAD DE TRATAMIENTO INTERMEDIO PEDIATRÍA                          5815
NEUROLOGÍA ADULTO                                                   5636
PSIQUIATRÍA CORTA ESTADÍA                                           5408
PSIQUIATRÍA                                                         5276
UNIDAD DE TRATAMIENTO INTERMEDIO MEDICINA ADULTO                    4932
UNIDAD EMERGENCIA ADULTOS                                           4866
PENSIONADO MATERNIDAD                                               4656
GERIATRÍA                                                           4063
AREA MEDICO-QUIRURGICOPEDIATRICA CUIDADOS MEDIOS                    4062
UNIDAD DE TRATAMIENTO INTERMEDIO CARDIOVASCULAR                     4008
UNIDAD DE AGUDOS MEDICO QUIRÚRGICOS                                 3723
AGUDOS CIRUGÍA                                                      3471
CIRUGÍA CARDIOVASCULAR                                              3154
OTORRINOLARINGOLOGÍA                                                3058
UNIDAD DE CUIDADOS INTENSIVOS (UCI) (INDIFERENCIADO)                3028
TRAUMATOLOGÍA Y ORTOPEDIA INFANTIL                                  2546
AREA MQ PEDIATRIA                                                   1970
AREA MEDICO-QUIRURGICOPEDIATRICA CUIDADOS BASICOS                   1906
UNIDAD DE AGUDOS NEUROLOGÍA                                         1853
CIRUGÍA DE PARTES BLANDAS INFANTIL                                  1823
UTI ADULTOS                                  

Existen 6534 filas que contienen DESCONOCIDO en la columna SERVICIO ALTA.
Al parecer Desconocido está como una opción y no representado como un valor faltante o nulo.

### Tabla manual con los resultados

In [27]:
# crear una tabla
import pandas as pd
data = {
    'Variable': ['Fecha de Nacimiento', 'Códigos de Procedimientos', 'Códigos GRD', 'Servicio de Alta'],
    'Problema': ['Valores nulos', 'Valores nulos en PROCEDIMIENTO1', 'Valores nulos donde GRD queda sin grupo', 'Valores "DESCONOCIDO"'],
    'Cantidad': [10, 172, 30, 6534],
    #'Posible Causa': ['Error de ingreso', 'Error desconocido', 'Problema sistemático en la recopilación de datos', 'Opción de registro'],
    'Acción Propuesta': ['Investigar y/o eliminar filas', 'Investigar, corregir o eliminar', 'Investigar y/o eliminar filas', 'Considerar como una categoría por la cantidad alta de datos']
}

df_analisis = pd.DataFrame(data)

#index desde 1
df_analisis.index = df_analisis.index + 1

df_analisis

Variable                                 Problema  \
1        Fecha de Nacimiento                            Valores nulos   
2  Códigos de Procedimientos          Valores nulos en PROCEDIMIENTO1   
3                Códigos GRD  Valores nulos donde GRD queda sin grupo   
4           Servicio de Alta                    Valores "DESCONOCIDO"   

   Cantidad                                   Acción Propuesta  
1        10                      Investigar y/o eliminar filas  
2       172                    Investigar, corregir o eliminar  
3        30                      Investigar y/o eliminar filas  
4      6534  Considerar como una categoría por la cantidad ...

In [28]:
# Definir el estilo de la tabla
estilos = [
    dict(selector="th", props=[("text-align", "center"), ("font-weight", "bold"), ("background-color", "#ADD8E6")]),
    dict(selector="td", props=[("text-align", "center"), ("background-color", "#e0f7fa")]),
    dict(selector="tr:nth-child(even)", props=[("background-color", "white")])
]

# Aplicar los estilos al dataframe
estilo_df = df_analisis.style.set_table_styles(estilos)

# Mostrar el dataframe
estilo_df

In [29]:
#crear un excel con la tabla estilo_df
estilo_df.to_excel('df_analisis.xlsx', index=False)

Esto representa una parte del análisis de valores nulos y acciones propuestas para continuar con el uso de los datos. Siendo un proceso iterativo, se puede volver a estos analisis.

## Post 3: Eliminación de filas y creación de columnas


### Eliminación de Valores Nulos

In [30]:
# Eliminar 10 filas con valores nulos en 'FECHA_NACIMIENTO'
df = df.dropna(subset=['FECHA_NACIMIENTO'])

# Eliminar 30 filas con valores nulos en 'IR_29301_COD_GRD'
df = df.dropna(subset=['IR_29301_COD_GRD'])

# Eliminar 172 las filas con valores nulos en 'PROCEDIMIENTO1'
df = df.dropna(subset=['PROCEDIMIENTO1'])

In [31]:
# Corroboramos con shape si es menor la cantidad de filas.
df.shape

(1039375, 130)

Shape indica Filas y Columnas

- Shape del inicio (1039587, 129)
- Shape actual despues de eliminar las filas (1039375, 130)

Al agregar la columna NOMRE_HOSPITAL ahora son 130 y no 129 las columnas totales del data set.

1039587 - 1039375 = 212.

Se eliminaron 212 filas.


### Creación de columnas

- EDAD
- ESTANCIA
- DES, descripción de GRD

#### Columna 'EDAD'

In [32]:
# Convertir 'FECHA_NACIMIENTO' a un objeto tiempo
df['FECHA_NACIMIENTO'] = pd.to_datetime(df['FECHA_NACIMIENTO'], format='%d-%m-%Y', errors='coerce')

# importar fecha y crear una variable que represente la fecha actual.
import datetime
today = datetime.date.today()

# Accede a la data en FECHA_NACIMIENTO y calcula la diferencia en días / 365
df['EDAD'] = (today - df['FECHA_NACIMIENTO'].dt.date).apply(lambda x: x.days) // 365

In [33]:
# Mover la columna 'FECHA_NACIMIENTO' a la derecha de la columna 'SEXO'
cols = list(df.columns)
cols.insert(cols.index('SEXO') + 1, cols.pop(cols.index('EDAD')))
df = df[cols]

In [34]:
df.head(2)

COD_HOSPITAL                                    NOMBRE_HOSPITAL  \
1        122100               Hospital Clínico Regional (Valdivia)   
2        114101  Complejo Hospitalario Dr. Sótero del Río (Sant...   

  CIP_ENCRIPTADO    SEXO  EDAD FECHA_NACIMIENTO ETNIA   PROVINCIA  \
1       72196813  HOMBRE    44       1980-04-13  OTRO    VALDIVIA   
2    DESCONOCIDO  HOMBRE    23       2001-11-04  OTRO  CORDILLERA   

        COMUNA                          NACIONALIDAD  \
1  PANGUIPULLI                                 CHILE   
2  PUENTE ALTO  VENEZUELA (REPÚBLICA BOLIVARIANA DE)   

                        PREVISION            SERVICIO_SALUD  \
1  FONASA INSTITUCIONAL - (MAI) A                  VALDIVIA   
2                      PARTICULAR  METROPOLITANO SURORIENTE   

                  TIPO_PROCEDENCIA TIPO_INGRESO ESPECIALIDAD_MEDICA  \
1  SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA     CIRUGÍA GENERAL   
2  SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA     CIRUGÍA GENERAL   

    TIPO_ACTIVIDAD FECHA_INGRESO               SERVICIOINGRESO FECHATRASLADO1  \
1  HOSPITALIZACIÓN    12-03-2023     TRAUMATOLOGÍA Y ORTOPEDIA            NaN   
2  HOSPITALIZACIÓN    04-11-2023  DERIVACIÓN MÉDICO QUIRÚRGICO     06-11-2023   

  SERVICIOTRASLADO1 FECHATRASLADO2 SERVICIOTRASLADO2 FECHATRASLADO3  \
1               NaN            NaN               NaN            NaN   
2           CIRUGÍA            NaN               NaN            NaN   

  SERVICIOTRASLADO3 FECHATRASLADO4 SERVICIOTRASLADO4 FECHATRASLADO5  \
1               NaN            NaN               NaN            NaN   
2               NaN            NaN               NaN            NaN   

  SERVICIOTRASLADO5 FECHATRASLADO6 SERVICIOTRASLADO6 FECHATRASLADO7  \
1               NaN            NaN               NaN            NaN   
2               NaN            NaN               NaN            NaN   

  SERVICIOTRASLADO7 FECHATRASLADO8 SERVICIOTRASLADO8 FECHATRASLADO9  \
1               NaN            NaN               NaN            NaN   
2               NaN            NaN               NaN            NaN   

  SERVICIOTRASLADO9   FECHAALTA               SERVICIOALTA   TIPOALTA  \
1               NaN  14-03-2023  TRAUMATOLOGÍA Y ORTOPEDIA  DOMICILIO   
2               NaN  10-11-2023                    CIRUGÍA  DOMICILIO   

   CONDICIONDEALTANEONATO1  PESORN1 SEXORN1  RN1ESTADO  \
1                      NaN      NaN     NaN        NaN   
2                      NaN      NaN     NaN        NaN   

   CONDICIONDEALTANEONATO2  PESORN2 SEXORN2  RN2ESTADO  \
1                      NaN      NaN     NaN        NaN   
2                      NaN      NaN     NaN        NaN   

   CONDICIONDEALTANEONATO3  PESORN3 SEXORN3  RN3ESTADO  \
1                      NaN      NaN     NaN        NaN   
2                      NaN      NaN     NaN        NaN   

   CONDICIONDEALTANEONATO4  PESORN4  SEXORN4  RN4ESTADO DIAGNOSTICO1  \
1                      NaN      NaN      NaN        NaN        K35.3   
2                      NaN      NaN      NaN        NaN        J93.1   

  DIAGNOSTICO2 DIAGNOSTICO3 DIAGNOSTICO4 DIAGNOSTICO5 DIAGNOSTICO6  \
1          NaN          NaN          NaN          NaN          NaN   
2          NaN          NaN          NaN          NaN          NaN   

  DIAGNOSTICO7 DIAGNOSTICO8 DIAGNOSTICO9 DIAGNOSTICO10 DIAGNOSTICO11  \
1          NaN          NaN          NaN           NaN           NaN   
2          NaN          NaN          NaN           NaN           NaN   

  DIAGNOSTICO12 DIAGNOSTICO13 DIAGNOSTICO14 DIAGNOSTICO15 DIAGNOSTICO16  \
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   

  DIAGNOSTICO17 DIAGNOSTICO18 DIAGNOSTICO19 DIAGNOSTICO20 DIAGNOSTICO21  \
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   

  DIAGNOSTICO22 DIAGNOSTICO23 DIAGNOSTICO24 DIAGNOSTICO25 DIAGNOSTICO

#### Columna ESTANCIA

In [35]:
# Limpieza y arreglo de datos
df['ESTANCIA'] = (pd.to_datetime(df['FECHAALTA'], format='%d-%m-%Y') - pd.to_datetime(df['FECHA_INGRESO'], format='%d-%m-%Y')).dt.days
df['ESTANCIA'] = df['ESTANCIA'].astype(int)


#### Columna DESC

Corresponde a la descripción del GRD correspondiente

Creación de la columna entre el cruce de IR_29301_COD_GRD y GRD_DESC

El Archivo excel GRD_DESC contiene una Hoja con dos columnas, una con el código numérico del GRD y otra con la descripción del GRD.

In [36]:
excel_file_path = 'GRD DESC.xlsx'

# cargar archivo excel con columna que contiene la descripción o nombre del GRD correspondiente
grd_desc = pd.read_excel(excel_file_path, sheet_name='Hoja1')

In [37]:
# cruzar IR_29301_COD_GRD de df con GRD de grd_desc para crear una columna en df que contenga DESC de grd_desc
df = pd.merge(df, grd_desc, left_on='IR_29301_COD_GRD', right_on='GRD', how='left')

In [38]:
# eliminar Columna GRD
df = df.drop('GRD', axis=1)

In [39]:
# Mover la columna DESC al lado derecho de IR_29301_PESO
column_to_move = df.pop('DESC')
df.insert(df.columns.get_loc('IR_29301_COD_GRD') + 1, 'DESC', column_to_move)

In [40]:
df.head(2)

COD_HOSPITAL                                    NOMBRE_HOSPITAL  \
0        122100               Hospital Clínico Regional (Valdivia)   
1        114101  Complejo Hospitalario Dr. Sótero del Río (Sant...   

  CIP_ENCRIPTADO    SEXO  EDAD FECHA_NACIMIENTO ETNIA   PROVINCIA  \
0       72196813  HOMBRE    44       1980-04-13  OTRO    VALDIVIA   
1    DESCONOCIDO  HOMBRE    23       2001-11-04  OTRO  CORDILLERA   

        COMUNA                          NACIONALIDAD  \
0  PANGUIPULLI                                 CHILE   
1  PUENTE ALTO  VENEZUELA (REPÚBLICA BOLIVARIANA DE)   

                        PREVISION            SERVICIO_SALUD  \
0  FONASA INSTITUCIONAL - (MAI) A                  VALDIVIA   
1                      PARTICULAR  METROPOLITANO SURORIENTE   

                  TIPO_PROCEDENCIA TIPO_INGRESO ESPECIALIDAD_MEDICA  \
0  SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA     CIRUGÍA GENERAL   
1  SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA     CIRUGÍA GENERAL   

    TIPO_ACTIVIDAD FECHA_INGRESO               SERVICIOINGRESO FECHATRASLADO1  \
0  HOSPITALIZACIÓN    12-03-2023     TRAUMATOLOGÍA Y ORTOPEDIA            NaN   
1  HOSPITALIZACIÓN    04-11-2023  DERIVACIÓN MÉDICO QUIRÚRGICO     06-11-2023   

  SERVICIOTRASLADO1 FECHATRASLADO2 SERVICIOTRASLADO2 FECHATRASLADO3  \
0               NaN            NaN               NaN            NaN   
1           CIRUGÍA            NaN               NaN            NaN   

  SERVICIOTRASLADO3 FECHATRASLADO4 SERVICIOTRASLADO4 FECHATRASLADO5  \
0               NaN            NaN               NaN            NaN   
1               NaN            NaN               NaN            NaN   

  SERVICIOTRASLADO5 FECHATRASLADO6 SERVICIOTRASLADO6 FECHATRASLADO7  \
0               NaN            NaN               NaN            NaN   
1               NaN            NaN               NaN            NaN   

  SERVICIOTRASLADO7 FECHATRASLADO8 SERVICIOTRASLADO8 FECHATRASLADO9  \
0               NaN            NaN               NaN            NaN   
1               NaN            NaN               NaN            NaN   

  SERVICIOTRASLADO9   FECHAALTA               SERVICIOALTA   TIPOALTA  \
0               NaN  14-03-2023  TRAUMATOLOGÍA Y ORTOPEDIA  DOMICILIO   
1               NaN  10-11-2023                    CIRUGÍA  DOMICILIO   

   CONDICIONDEALTANEONATO1  PESORN1 SEXORN1  RN1ESTADO  \
0                      NaN      NaN     NaN        NaN   
1                      NaN      NaN     NaN        NaN   

   CONDICIONDEALTANEONATO2  PESORN2 SEXORN2  RN2ESTADO  \
0                      NaN      NaN     NaN        NaN   
1                      NaN      NaN     NaN        NaN   

   CONDICIONDEALTANEONATO3  PESORN3 SEXORN3  RN3ESTADO  \
0                      NaN      NaN     NaN        NaN   
1                      NaN      NaN     NaN        NaN   

   CONDICIONDEALTANEONATO4  PESORN4  SEXORN4  RN4ESTADO DIAGNOSTICO1  \
0                      NaN      NaN      NaN        NaN        K35.3   
1                      NaN      NaN      NaN        NaN        J93.1   

  DIAGNOSTICO2 DIAGNOSTICO3 DIAGNOSTICO4 DIAGNOSTICO5 DIAGNOSTICO6  \
0          NaN          NaN          NaN          NaN          NaN   
1          NaN          NaN          NaN          NaN          NaN   

  DIAGNOSTICO7 DIAGNOSTICO8 DIAGNOSTICO9 DIAGNOSTICO10 DIAGNOSTICO11  \
0          NaN          NaN          NaN           NaN           NaN   
1          NaN          NaN          NaN           NaN           NaN   

  DIAGNOSTICO12 DIAGNOSTICO13 DIAGNOSTICO14 DIAGNOSTICO15 DIAGNOSTICO16  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

  DIAGNOSTICO17 DIAGNOSTICO18 DIAGNOSTICO19 DIAGNOSTICO20 DIAGNOSTICO21  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

  DIAGNOSTICO22 DIAGNOSTICO23 DIAGNOSTICO24 DIAGNOSTICO25 DIAGNOSTICO

### Cambiar el tipo de dato en columnas

In [41]:
# Convertir las columnas al tipo integer
for col in [ 'IR_29301_SEVERIDAD', 'IR_29301_MORTALIDAD']:
    df[col] = pd.to_numeric(df[col], errors='coerce').astype('Int64')

Tipo de datos en DIAGNOSTICO1 y PROCEDIMIENTO1

In [42]:
# Muestra los tipos de datos para las columnas seleccionadas
print(df['DIAGNOSTICO1'].dtype)
print(df['PROCEDIMIENTO1'].dtype)
print(df['IR_29301_COD_GRD'].dtype)
print(df['IR_29301_PESO'].dtype)
print(df['IR_29301_SEVERIDAD'].dtype)
print(df['IR_29301_MORTALIDAD'].dtype)

object
float64
float64
object
Int64
Int64


In [43]:
# Remover el .0 de todas las columnas de PROCEDIMIENTO
for i in range(1, 31):
    col_name = f'PROCEDIMIENTO{i}'
    if col_name in df.columns:

        df[col_name] = df[col_name].astype(str)
        df[col_name] = df[col_name].str.replace('.0', '', regex=False)

In [44]:
# pasar IR_29301_COD_GRD a numero entero
df['IR_29301_COD_GRD'] = df['IR_29301_COD_GRD'].astype(int)

In [45]:
# transformar los NaN de USOPABELLON a 0
df['USOSPABELLON'] = df['USOSPABELLON'].fillna(0)

In [46]:
# pasar USOSPABELLON a numero entero
df['USOSPABELLON'] = df['USOSPABELLON'].astype(int)

In [47]:
# cambiar la , por . en IR_29301_PESO
df['IR_29301_PESO'] = df['IR_29301_PESO'].str.replace(',', '.')

In [48]:
# pasar IR_29301_PESO a float
df['IR_29301_PESO'] = df['IR_29301_PESO'].astype(float)

In [49]:
# guardar df_final como csv
df.to_csv('df_final.csv', index=False)

Crear df sólo con los casos de Hospital de Quilpué en NOMBRE_HOSPITAL

In [53]:
# df_quilpue = df NOMBRE_HOSPITAL == hospital a eleccion
# En este caso elegiremos los datos del Instituto de Neurocirugía Dr. Alfonso Asenjo, por ser el que menor cantidad de egresos presenta
df_hospital_eleccion = df[df['NOMBRE_HOSPITAL'] == 'Instituto de Neurocirugía Dr. Alfonso Asenjo']

# Se puede reemplazar el nombre del hospital para obtener los datos de otro.

In [59]:
# index reiniciar df_hospital_eleccion desde 1
df_hospital_eleccion.index = df_hospital_eleccion.index + 1

In [60]:
# df_hospital_eleccion a csv
df_hospital_eleccion.to_csv('df_hospital_eleccion.csv', index=False)

In [61]:
df_hospital_eleccion.head()

COD_HOSPITAL                               NOMBRE_HOSPITAL CIP_ENCRIPTADO  \
1        112104  Instituto de Neurocirugía Dr. Alfonso Asenjo       70173725   
2        112104  Instituto de Neurocirugía Dr. Alfonso Asenjo       83826897   
3        112104  Instituto de Neurocirugía Dr. Alfonso Asenjo      100057782   
4        112104  Instituto de Neurocirugía Dr. Alfonso Asenjo       79732091   
5        112104  Instituto de Neurocirugía Dr. Alfonso Asenjo       87927204   

     SEXO  EDAD FECHA_NACIMIENTO ETNIA    PROVINCIA       COMUNA NACIONALIDAD  \
1  HOMBRE    55       1969-11-05  OTRO     SANTIAGO    LO ESPEJO        CHILE   
2   MUJER    47       1977-11-04  OTRO     SANTIAGO    PEÑALOLÉN        CHILE   
3  HOMBRE    36       1989-01-01  OTRO     SANTIAGO     SANTIAGO        CHILE   
4  HOMBRE    43       1981-07-31  OTRO  ANTOFAGASTA  ANTOFAGASTA        CHILE   
5  HOMBRE    54       1970-09-08  OTRO     SANTIAGO        ÑUÑOA        CHILE   

                        PREVISION         SERVICIO_SALUD  \
1  FONASA INSTITUCIONAL - (MAI) C      METROPOLITANO SUR   
2  FONASA INSTITUCIONAL - (MAI) B  METROPOLITANO ORIENTE   
3  FONASA INSTITUCIONAL - (MAI) A  METROPOLITANO CENTRAL   
4  FONASA INSTITUCIONAL - (MAI) D            ANTOFAGASTA   
5  FONASA INSTITUCIONAL - (MAI) B  METROPOLITANO ORIENTE   

                                    TIPO_PROCEDENCIA TIPO_INGRESO  \
1  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
2  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
3                      OTROS HOSPITALES RED NACIONAL     URGENCIA   
4  CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...   PROGRAMADA   
5                    SERVICIO EMERGENCIA (DOMICILIO)     URGENCIA   

  ESPECIALIDAD_MEDICA   TIPO_ACTIVIDAD FECHA_INGRESO  \
1    NEURORRADIOLOGIA  HOSPITALIZACIÓN    02-08-2023   
2        NEUROCIRUGÍA  HOSPITALIZACIÓN    31-07-2023   
3        NEUROCIRUGÍA  HOSPITALIZACIÓN    20-09-2023   
4        NEUROCIRUGÍA  HOSPITALIZACIÓN    24-07-2023   
5        NEUROCIRUGÍA  HOSPITALIZACIÓN    26-06-2023   

                                    SERVICIOINGRESO FECHATRASLADO1  \
1                               NEUROCIRUGÍA ADULTO            NaN   
2  UNIDAD DE TRATAMIENTO INTERMEDIO NEUROQUIRÚRGICO            NaN   
3  UNIDAD DE TRATAMIENTO INTERMEDIO NEUROQUIRÚRGICO     20-09-2023   
4                               NEUROCIRUGÍA ADULTO     25-07-2023   
5  UNIDAD DE TRATAMIENTO INTERMEDIO NEUROQUIRÚRGICO     28-06-2023   

                                  SERVICIOTRASLADO1 FECHATRASLADO2  \
1                                               NaN            NaN   
2                                               NaN            NaN   
3  UNIDAD DE TRATAMIENTO INTERMEDIO NEUROQUIRÚRGICO     21-09-2023   
4  UNIDAD DE TRATAMIENTO INTERMEDIO NEUROQUIRÚRGICO     25-07-2023   
5                               NEUROCIRUGÍA ADULTO     17-07-2023   

                                  SERVICIOTRASLADO2 FECHATRASLADO3  \
1                                               NaN            NaN   
2                                               NaN            NaN   
3                               NEUROCIRUGÍA ADULTO            NaN   
4                               NEUROCIRUGÍA ADULTO            NaN   
5  UNIDAD DE TRATAMIENTO INTERMEDIO NEUROQUIRÚRGICO     17-07-2023   

     SERVICIOTRASLADO3 FECHATRASLADO4 SERVICIOTRASLADO4 FECHATRASLADO5  \
1                  NaN            NaN               NaN            NaN   
2                  NaN            NaN               NaN            NaN   
3                  NaN            NaN               NaN            NaN   
4                  NaN            NaN               NaN            NaN   
5  NEUROCIRUGÍA ADULTO            NaN               NaN            NaN   

  SERVICIOTRASLADO5 FECHATRASLADO6 SERVICIOTRASLADO6 FECHATRASLADO7  \
1               NaN            NaN               NaN            NaN   
2               NaN            NaN               NaN            NaN   
3

In [65]:
df.shape


(1039375, 133)